starting from another folder issue

hey there, great work on the driver, i ran into a wall when making a scoop json file
it seems that the ps scripts dont like to run outside of their folder

FYI: with scoop you can do something like `scoop install cxadc-win ffmpeg flac sox curl` on windows

```
`PS C:\Users\Tommy> localcapture.ps1 -N test6 -video 0
Invoke-Expression: Unexpected token 'unix:C:\Users\Tommy\.capture-server.sock' in expression or statement.
LocalCapture.ps1: server not responding
Waiting for writes to finish...
Files created:
Killing server
Finished!`
```

cx device is available
```
`PS C:\Users\Tommy> Get-CxadcWinDevice

Path       InstanceName                                                           Location
----       ------------                                                           --------
\\.\cxadc0 PCI\VEN_14F1&DEV_8800&SUBSYS_45534E4F&REV_05\7&1afcafbc&0&0000300211_0 35:00.0`
```

in the folder it seem to run fine
```
`PS C:\Users\Tommy\scoop\apps\cxadc-win\current\bin> .\LocalCapture.ps1  -n test -video 0
Starting "test" capture, press 'q' to stop
Waiting for writes to finish...
Files created:
  test-video.u8
Killing server`

```

<img width="985" height="756" alt="Image" src="https://github.com/user-attachments/assets/d03448cd-c59e-406b-9d04-7dc383b3137e" />

scoop json

```
{
    "version": "1.1",
    "description": "Open source driver for CX23880/1/2/3 PCIe video capture cards, including capture tools and scripts.",
    "homepage": "https://github.com/JuniorIsAJitterbug/cxadc-win",
    "license": "GPL-2.0-only",
    "notes": [
        "1. This package requires Administrator privileges to install the driver.",
        "2. IMPORTANT: You must reboot after enabling Test Mode for the driver to load.",
        "3. To load the PowerShell modules in your session, run:",
        "   Import-Module \"$(scoop prefix cxadc-win)\\bin\\CxadcWin.psm1\" -Force",
        "   Import-Module \"$(scoop prefix cxadc-win)\\bin\\CxadcClockGen.psm1\" -Force"
    ],
    "architecture": {
        "64bit": {
            "url": "https://github.com/JuniorIsAJitterbug/cxadc-win/releases/download/v1.1/cxadc-win-v1.1-x64.zip",
            "hash": "1c4e880371327fdc6fd18817ae21166716a1120fb86c613f8f8f2c9cc9ff4e9e"
        }
    },
    "env_add_path": "bin",
    "bin": [
        ["enable-cxadc-testmode.ps1", "enable-cxadc-testmode"],
        "bin\\leveladj.exe",
        "bin\\capture-server.exe",
        "bin\\LocalCapture.ps1"
    ],
    "installer": {
        "script": [
            "if (!(is_admin)) { error \"Administrator rights are required to install drivers. Please run with 'sudo' or in an Admin terminal.\"; break }",
            "",
            "# --- 1. Generate Helper Script for Test Mode ---",
            "Write-Host \"Creating helper script 'enable-cxadc-testmode'...\" -ForegroundColor Cyan",
            "$helperPath = \"$dir\\enable-cxadc-testmode.ps1\"",
            "$helperContent = @(",
            "    'if (!([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) { Write-Error \"Administrator privileges required. Run: sudo enable-cxadc-testmode\"; exit 1 }',",
            "    'Write-Host \"Attempting to enable Test Signing...\" -ForegroundColor Cyan',",
            "    '$res = bcdedit /set testsigning on 2>&1',",
            "    'if ($LASTEXITCODE -eq 0) { Write-Host \"[SUCCESS] Test Mode enabled. Please REBOOT your computer.\" -ForegroundColor Green }',",
            "    'else { Write-Host \"[ERROR] Failed to enable Test Mode.\" -ForegroundColor Red; Write-Host $res; if ($res -match \"Secure Boot\") { Write-Host \"HINT: Disable Secure Boot in BIOS.\" -ForegroundColor Yellow } }'",
            ")",
            "Set-Content -Path $helperPath -Value $helperContent",
            "",
            "# --- 2. Check Requirements (Color Coded) ---",
            "Write-Host \"Checking system requirements...\" -ForegroundColor Cyan",
            "try {",
            "    if (Confirm-SecureBootUEFI) {",
            "        Write-Host \"[WARNING] Secure Boot is ENABLED. The driver will fail to load.\" -ForegroundColor Red",
            "    } else {",
            "        Write-Host \"[OK] Secure Boot is DISABLED.\" -ForegroundColor Green",
            "    }",
            "} catch {}",
            "",
            "$bcd = bcdedit /enum \"{current}\" | Out-String",
            "if ($bcd -match \"testsigning\\s+Yes\") {",
            "    Write-Host \"[OK] Test Mode is ENABLED.\" -ForegroundColor Green",
            "} else {",
            "    Write-Host \"[WARNING] Test Mode is OFF. Driver will not load until enabled.\" -ForegroundColor Red",
            "    Write-Host \"    Run 'sudo enable-cxadc-testmode' to fix this.\" -ForegroundColor Magenta",
            "}",
            "",
            "# --- 3. Install Driver via PNPUtil ---",
            "Write-Host \"Installing cxadc-win driver...\" -ForegroundColor Cyan",
            "pnputil /add-driver \"$dir\\driver\\cxadc-win.inf\" /install"
        ]
    },
    "uninstaller": {
        "script": [
            "if (!(is_admin)) { error \"Administrator rights are required to uninstall drivers.\"; break }",
            "Write-Host \"Uninstalling cxadc-win driver...\" -ForegroundColor Cyan",
            "pnputil /delete-driver \"$dir\\driver\\cxadc-win.inf\" /force"
        ]
    },
    "checkver": {
        "url": "https://github.com/JuniorIsAJitterbug/cxadc-win/releases",
        "regex": "/releases/tag/v([\\d.]+)"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/JuniorIsAJitterbug/cxadc-win/releases/download/v$version/cxadc-win-v$version-x64.zip"
            }
        }
    }
}``

--- a/src/capture-server/LocalCapture.ps1
+++ b/src/capture-server/LocalCapture.ps1
@@ -208,16 +208,26 @@ class BasebandDeviceData : DeviceData {
     [ValidateNotNullOrEmpty()][boolean] $EnableConversion
 }
 
+class EscapedPath {
+    [string] $Path
+    [string] $EscapedPath
+
+    EscapedPath([string] $Path) {
+        $this.Path = $Path
+        $this.EscapedPath = ("`"" + $Path + "`"")
+    }
+}
+
 class BinaryPaths {
-    [ValidateNotNullOrEmpty()][string] $Server
-    [ValidateNotNullOrEmpty()][string] $Curl
-    [string] $Flac
-    [string] $Sox
-    [string] $FFmpeg
+    [ValidateNotNullOrEmpty()][EscapedPath] $Server
+    [ValidateNotNullOrEmpty()][EscapedPath] $Curl
+    [EscapedPath] $Flac
+    [EscapedPath] $Sox
+    [EscapedPath] $FFmpeg
 }
 
 Class CxCaptureServer {
-    [string] $Socket
+    [EscapedPath] $Socket
     [BinaryPaths] $BinaryPaths
     [string] $BaseUrl = "http://capture-server"
     [boolean] $IsRunning = $false
@@ -225,7 +235,7 @@ Class CxCaptureServer {
     $ServerJob
     $CaptureJobs = @()
 
-    CxCaptureServer([string] $Socket, [BinaryPaths] $BinaryPaths) {
+    CxCaptureServer([EscapedPath] $Socket, [BinaryPaths] $BinaryPaths) {
         $this.Socket = $Socket
         $this.BinaryPaths = $BinaryPaths
     }
@@ -233,8 +243,8 @@ Class CxCaptureServer {
     [void] StartLocalServer() {
         $JobName = "ServerJob"
         $Command = @(
-            $this.BinaryPaths.Server,
-            ("unix:" + $this.Socket)
+            "&", $this.BinaryPaths.Server.EscapedPath,
+            ("unix:" + $this.Socket.EscapedPath)
         ) -join " "
 
         $this.ServerJob = Start-Job -Name $JobName -ScriptBlock {
@@ -248,7 +258,7 @@ Class CxCaptureServer {
     }
 
     [boolean] TestConnection() {
-        if ((Invoke-WebRequest -UnixSocket $this.Socket -Method Get $this.BaseUrl).StatusCode -eq 200) {
+        if ((Invoke-WebRequest -UnixSocket $this.Socket.Path -Method Get $this.BaseUrl).StatusCode -eq 200) {
             return $true
         }
 
@@ -257,7 +267,7 @@ Class CxCaptureServer {
 
     [int] GetOverflows() {
         $Url = $this.BaseUrl + "/stats"
-        $Json = Invoke-WebRequest -Method Get -UnixSocket $this.Socket $Url | ConvertFrom-Json
+        $Json = Invoke-WebRequest -Method Get -UnixSocket $this.Socket.Path $Url | ConvertFrom-Json
         return $Json.overflows
     }
 
@@ -273,7 +283,7 @@ Class CxCaptureServer {
         }
 
         $Url = $this.BaseUrl + "/start?" + ($Params -Join '&')
-        $Json = Invoke-WebRequest -Method Get -UnixSocket $this.Socket $Url | ConvertFrom-Json
+        $Json = Invoke-WebRequest -Method Get -UnixSocket $this.Socket.Path $Url | ConvertFrom-Json
 
         if ($Json.state -eq "Running") {
             $this.IsRunning = $true
@@ -293,7 +303,7 @@ Class CxCaptureServer {
         }
 
         $Url = $this.BaseUrl + "/stop"
-        $Json = Invoke-WebRequest -Method Get -UnixSocket $this.Socket $Url | ConvertFrom-Json
+        $Json = Invoke-WebRequest -Method Get -UnixSocket $this.Socket.Path $Url | ConvertFrom-Json
 
         if ($Json.state -ne "Idle") {
             throw "unable to stop server"
@@ -312,9 +322,9 @@ Class CxCaptureServer {
     [void] CaptureCx([CxDeviceData] $Device) {
         $JobName = "CaptureCxJob" + $Device.Index
         $CurlCommand = @(
-            "&", $this.BinaryPaths.Curl, "-s",
+            "&", $this.BinaryPaths.Curl.EscapedPath, "-s",
             "-X", "GET",
-            "--unix-socket", $this.Socket,
+            "--unix-socket", $this.Socket.EscapedPath,
             "--url", ($this.BaseUrl + "/cxadc?" + $Device.Index),
             "-o"
         )
@@ -328,7 +338,7 @@ Class CxCaptureServer {
             $FlacOut = ($Device.FilePrefix + ".flac")
             $FlacRate = ($Device.EnableResampling) ? $Device.ResampleRate : $Device.Rate
             $FlacCommand = @(
-                "|", "&", $this.BinaryPaths.Flac, "-s", "-f",
+                "|", "&", $this.BinaryPaths.Flac.EscapedPath, "-s", "-f",
                 "-j", $Device.FlacThreadCount
                 ("-" + $Device.CompressionLevel), "-b", 65535, "--lax",
                 ("--sample-rate=" + $FlacRate), "--channels=1", "--bps=8", "--sign=unsigned", "--endian=little",
@@ -345,7 +355,7 @@ Class CxCaptureServer {
 
             if ($Device.UseSox) {
                 $ResamplerCommand = @(
-                    "|", "&", $this.BinaryPaths.Sox,
+                    "|", "&", $this.BinaryPaths.Sox.EscapedPath,
                     "-D",
                     "-t", "raw", "-r", $Device.Rate, "-b", 8, "-c", 1, "-L", "-e", "unsigned-integer",
                     "-",
@@ -354,7 +364,7 @@ Class CxCaptureServer {
                 )
             } else {
                 $ResamplerCommand = @(
-                    "|", "&", $this.BinaryPaths.FFmpeg,
+                    "|", "&", $this.BinaryPaths.FFmpeg.EscapedPath,
                     "-hide_banner", "-y", "-loglevel", "error",
                     "-threads", $Device.FFmpegThreadCount,
                     "-thread_queue_size", 1024,
@@ -385,9 +395,9 @@ Class CxCaptureServer {
         $JobName = "CaptureBasebandJob"
         $CurlOut = ($Device.FilePrefix + ".s24")
         $CurlCommand = @(
-            "&", $this.BinaryPaths.Curl, "-s",
+            "&", $this.BinaryPaths.Curl.EscapedPath, "-s",
             "-X", "GET",
-            "--unix-socket", $this.Socket,
+            "--unix-socket", $this.Socket.EscapedPath,
             "--url", ($this.BaseUrl + "/baseband"),
             "-o"
         )
@@ -400,7 +410,7 @@ Class CxCaptureServer {
             $FFmpegOutHeadSwitchFileType = (($Device.CompressHeadSwitch) ? "flac" : "u8")
             $FFmpegOutHeadSwitch = ($Device.HeadSwitchFileName + "." + $FFmpegOutHeadSwitchFileType)
             $FFmpegCommand = @(
-                "|", "&", $this.BinaryPaths.FFmpeg,
+                "|", "&", $this.BinaryPaths.FFmpeg.EscapedPath,
                 "-hide_banner", "-y", "-loglevel", "error",
                 "-threads", $Device.FFmpegThreadCount,
                 "-ar", $Device.Rate, "-ac", 3, "-f", "s24le",
@@ -498,17 +508,24 @@ if ($Devices.Count -eq 0) {
 $FindBinary = {
     param([string] $Name)
 
-    $Fullname = ($Name + ".exe")
-    $TCResult = (Test-Path (".\" + $Fullname))
+    $FileName = ($Name + ".exe")
+    $ScriptPath = (Join-Path -Path $PSScriptRoot -ChildPath $FileName)
 
-    if ($TCResult -eq $true) {
-        return (".\`"" + $Fullname + "`"")
+    # check .\
+    if ((Test-Path -Path $FileName) -eq $true) {
+        return (Get-Item -Path $FileName).FullName
     }
 
-    $GCResult = (Get-Command $Fullname -ErrorAction SilentlyContinue)
+    # check script path
+    if ((Test-Path -Path $ScriptPath) -eq $true) {
+        return $ScriptPath
+    }
+
+    # check $PATH
+    $GCResult = (Get-Command -Name $FileName -ErrorAction SilentlyContinue)
 
     if ($GCResult -ne $null) {
-        return ("`"" + $GCResult.Path + "`"")
+        return $GCResult.Path
     }
 
     throw ($Name + " not found in path")
@@ -517,27 +534,27 @@ $FindBinary = {
 # check path for binaries
 try {
     $BinaryPaths = [BinaryPaths]@{
-        Server = $FindBinary.Invoke("capture-server")
-        Curl = $FindBinary.Invoke("curl")
+        Server = [EscapedPath]::new($FindBinary.Invoke("capture-server"))
+        Curl = [EscapedPath]::new($FindBinary.Invoke("curl"))
     }
 
     if ($CompressVideo -or $CompressHifi) {
-        $BinaryPaths.Flac = $FindBinary.Invoke("flac")
+        $BinaryPaths.Flac = [EscapedPath]::new($FindBinary.Invoke("flac"))
     }
 
     if (($ResampleVideo -or $ResampleHifi) -and $UseSox) {
-        $BinaryPaths.Sox = $FindBinary.Invoke("sox")
+        $BinaryPaths.Sox = [EscapedPath]::new($FindBinary.Invoke("sox"))
     }
 
     if ($Baseband -or (($ResampleVideo -or $ResampleHifi) -and !$UseSox)) {
-        $BinaryPaths.FFmpeg = $FindBinary.Invoke("ffmpeg")
+        $BinaryPaths.FFmpeg = [EscapedPath]::new($FindBinary.Invoke("ffmpeg"))
     }
 } catch {
     Write-Error $_
     exit
 }
 
-$SocketPath = ((Get-Location).Path + "\.capture-server.sock")
+$SocketPath = [EscapedPath]::new((Join-Path -Path (Get-Location).Path -ChildPath "\.capture-server.sock"))
 $CxCaptureServer = [CxCaptureServer]::new($SocketPath, $BinaryPaths)
 
 try {

